ddtrace/tracer/remote_config.go

```diff
@@ -18,6 +20,38 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
 )
 
+type dynamicInstrumentationRCProbeConfig struct {
+	runtimeID     string
+	configPath    string
+	configContent string
+}
+
+type dynamicInstrumentationRCState struct {
+	mu    *sync.Mutex
+	state map[string]dynamicInstrumentationRCProbeConfig
+}
+
+var (
+	diRCState dynamicInstrumentationRCState
+)
+
+func init() {
+	diRCState = dynamicInstrumentationRCState{
+		mu: &sync.Mutex{},
+	}
+
+	go func() {
+		for {
+			time.Sleep(time.Second * 5)
+			diRCState.mu.Lock()
+			for _, v := range diRCState.state {
+				passFullConfiguration(v)
+			}
+			diRCState.mu.Unlock()
+		}
+	}()
+}
```

Comment:
I feel this belongs to `tracer.startRemoteConfig`. You can use a `sync.Once` to avoid running it multiple times if you want to be extra cautious.

Also, I think we should use `R{,Un}Lock` instead.

Is there any reason it should be initialized even before the tracer is created?

Reply:
Another good call on moving out of an init, I've moved the logic into `startRemoteConfig()`. Also i've cleaned up the placement of the DI specific types/functions in this file. 

I don't see a downside to use a RWMutex but there's only one reader, the looping goroutine calling `passFullConfiguration` (which i've changed to `passProbeConfiguration` to be more accurate).